index.js

```diff
@@ -51,49 +55,106 @@ const selectors = {
 };
 
 class Scholar {
-  init(key) {
-    if (!key) return this;
-    this.isProxy = true;
-    client.defaults.options = got.mergeOptions(client.defaults.options, {
+  init(apiKey, { retries } = {}) {
+    const useProxy = !apiKey;
+    this.useProxy = useProxy;
+
+    let client = this._setupClient();
+    if (useProxy) client = this._setupProxy(client, apiKey);
+    this.client = client;
+
+    if (retries === undefined) retries = useProxy ? 2 : 0;
+    this.retries = Math.min(0, retries);
```

Comment:
I don't get why you are returning min of zero and `retries`? For example, if `retries = 2`, `this.retries` will become 0 ?

Reply:
Because I'm an idiot. What I've should've done is use `Math.max` to ensure no negative retry count could be set.

Reply:
```suggestion
    this.retries = Math.max(0, retries);
```